cmd/manager/main.go

```diff
@@ -47,8 +96,17 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Delete incompatible ConfigMap lock if it exists
+	lockName := "cluster-storage-operator-lock"
+	err = tryDeleteIncompatibleLock(cfg, lockName)
+	if err != nil {
+		// We don't exit if an error happens, but we let the operator
+		// try to become the leader below. If it doesn't succeed, then we exit
```

Comment:
In https://bugzilla.redhat.com/show_bug.cgi?id=1877316 we saw 4.5 CSO trying to become leader for 3 hours without exiting. There should be either timeout for `leader.Become` or CSO should exit here.

Reply:
I was expecting that on line 109 below it would `os.Exit(1)` if CSO couldn't become the leader, but operator-sdk doesn't return an error the ConfigMap already exists:

https://github.com/bertinatto/cluster-storage-operator/blob/1b1fae52bda9667b7b9bf7da7397c88a82faa4cc/vendor/github.com/operator-framework/operator-sdk/pkg/leader/leader.go#L124-L131

Reply:
Fixed, PTAL